src/psycop/projects/t2d/utils/pipeline_objects.py

```diff
@@ -82,62 +114,103 @@ def cfg(self) -> FullConfigSchema:
-        df = pd.read_parquet(self.eval_dir / "evaluation_dataset.parquet")
+        df = pd.read_parquet(self.dir_path / "evaluation_dataset.parquet")
 
         eval_dataset = df_to_eval_dataset(df, custom_columns=custom_columns)
 
         return eval_dataset
 
     def get_auroc(self) -> float:
         df = self.group.all_runs_performance_df
-
         self_run = df[df["run_name"] == self.name]
-
         return self_run["roc_auc"].iloc[0]
 
     @property
     def pipe(self) -> Pipeline:
-        return load_file_from_pkl(self.eval_dir / "pipe.pkl")
+        return load_file_from_pkl(self.dir_path / "pipe.pkl")
 
 
-def load_file_from_pkl(file_path: Path) -> Any:
-    with file_path.open("rb") as f:
-        return pickle.load(f)
+@dataclass
+class T2DArtifactNames:
+    main_performance_figure: str = "t2d_main_performance_figure.png"
+    main_robustness_figure: str = "t2d_main_robustness.png"
+    performance_by_ppr: str = "t2d_performance_by_ppr.xlsx"
 
 
-def df_to_eval_dataset(
-    df: pd.DataFrame,
-    custom_columns: Optional[Sequence[str]],
-) -> EvalDataset:
-    """Convert dataframe to EvalDataset."""
-    return EvalDataset(
-        ids=df["ids"],
-        y=df["y"],
-        y_hat_probs=df["y_hat_probs"],
-        pred_timestamps=df["pred_timestamps"],
-        outcome_timestamps=df["outcome_timestamps"],
-        age=df["age"],
-        is_female=df["is_female"],
-        exclusion_timestamps=df["exclusion_timestamps"],
-        pred_time_uuids=df["pred_time_uuids"],
-        custom_columns={col: df[col] for col in custom_columns}
-        if custom_columns
-        else None,
-    )
+class PaperOutputPaths:
+    def __init__(self, artifact_path: Path, create_output_paths_on_init: bool = True):
+        self.artifact = artifact_path
+        self.tables = self.artifact / "tables"
+        self.figures = self.artifact / "figures"
+        self.estimates = self.artifact / "estimates"
+
+        if create_output_paths_on_init:
+            for path in [self.artifact, self.tables, self.figures, self.estimates]:
+                path.mkdir(parents=True, exist_ok=True)
+
+
+class PaperOutputSettings:
+    def __init__(
+        self,
+        name: str,
+        pos_rate: float,
+        model_type: str,
+        lookahead_days: int,
+        artifact_root: Optional[Path] = None,
+        create_output_paths_on_init: bool = True,
+    ):
+        self.name = name
+        self.pos_rate = pos_rate
+        artifact_root = EVAL_ROOT if artifact_root is None else artifact_root
+        self.artifact_path = (
+            artifact_root / f"{lookahead_days}_{model_type}_{self.name}"
+        )
+        self.artifact_names = T2DArtifactNames()
+        self.paths = PaperOutputPaths(
+            self.artifact_path,
+            create_output_paths_on_init=create_output_paths_on_init,
+        )
+
+
+class PipelineRun:
+    def __init__(
+        self,
+        name: str,
+        group: RunGroup,
+        pos_rate: float,
```

Comment:
Perhaps we want to add this abstraction to the library? It's strictly coupled to the folder structure output by `train_model`, so if we want it, we need a contract-test that ensures that the interface between the two is stable.